pkg/crdpuller/discovery.go

```diff
@@ -92,133 +107,161 @@ func (sp *schemaPuller) PullCRDs(context context.Context, resourceNames ...strin
+					continue
 				}
+			}
 
-				scaleSubResource := &apiextensionsv1.CustomResourceSubresourceScale{
-					SpecReplicasPath:   ".spec.replicas",
-					StatusReplicasPath: ".status.replicas",
-				}
-				if !hasSubResource("scale") {
-					scaleSubResource = nil
+			hasSubResource := func(subResource string) bool {
+				groupResourceNames := apiResourceNames[gv]
+				if groupResourceNames != nil {
+					return groupResourceNames.Has(apiResource.Name + "/" + subResource)
 				}
+				return false
+			}
+
+			statusSubResource := &apiextensionsv1.CustomResourceSubresourceStatus{}
+			if !hasSubResource("status") {
+				statusSubResource = nil
+			}
 
-				crd = &apiextensionsv1.CustomResourceDefinition{
-					TypeMeta:   typeMeta,
-					ObjectMeta: objectMeta,
-					Spec: apiextensionsv1.CustomResourceDefinitionSpec{
-						Group: gv.Group,
-						Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
-							{
-								Name: gv.Version,
-								Schema: &apiextensionsv1.CustomResourceValidation{
-									OpenAPIV3Schema: &schemaProps,
-								},
-								Subresources: &apiextensionsv1.CustomResourceSubresources{
-									Status: statusSubResource,
-									Scale:  scaleSubResource,
-								},
-								Served:  true,
-								Storage: true,
+			scaleSubResource := &apiextensionsv1.CustomResourceSubresourceScale{
+				SpecReplicasPath:   ".spec.replicas",
+				StatusReplicasPath: ".status.replicas",
+			}
+			if !hasSubResource("scale") {
+				scaleSubResource = nil
+			}
+
+			crd = &apiextensionsv1.CustomResourceDefinition{
+				TypeMeta:   typeMeta,
+				ObjectMeta: objectMeta,
+				Spec: apiextensionsv1.CustomResourceDefinitionSpec{
+					Group: gv.Group,
+					Versions: []apiextensionsv1.CustomResourceDefinitionVersion{
+						{
+							Name: gv.Version,
+							Schema: &apiextensionsv1.CustomResourceValidation{
+								OpenAPIV3Schema: &schemaProps,
 							},
-						},
-						Scope: resourceScope,
-						Names: apiextensionsv1.CustomResourceDefinitionNames{
-							Plural:     apiResource.Name,
-							Kind:       apiResource.Kind,
-							Categories: apiResource.Categories,
-							ShortNames: apiResource.ShortNames,
-							Singular:   apiResource.SingularName,
+							Subresources: &apiextensionsv1.CustomResourceSubresources{
+								Status: statusSubResource,
+								Scale:  scaleSubResource,
+							},
+							Served:  true,
+							Storage: true,
 						},
 					},
-				}
-
-				apiextensionsv1.SetDefaults_CustomResourceDefinition(crd)
-				crds[apiResource.Name] = crd
+					Scope: resourceScope,
+					Names: apiextensionsv1.CustomResourceDefinitionNames{
+						Plural:     apiResource.Name,
+						Kind:       apiResource.Kind,
+						Categories: apiResource.Categories,
+						ShortNames: apiResource.ShortNames,
+						Singular:   apiResource.SingularName,
+					},
+				},
+			}
 
-				break
+			apiextensionsv1.SetDefaults_CustomResourceDefinition(crd)
+			if apihelpers.IsProtectedCommunityGroup(gv.Group) {
```

Comment:
Add a comment describing why we're doing this here, and how it's related to that PR.